sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessorBuilder.java

```diff
@@ -154,8 +156,13 @@ int getMaxExportBatchSize() {
    * forwards them to the given {@code spanExporter}.
    *
    * @return a new {@link BatchSpanProcessor}.
+   * @throws IllegalArgumentException if {@code maxExportBatchSize} is greater than {@code
+   *     maxQueueSize}.
    */
   public BatchSpanProcessor build() {
+    checkArgument(
+        maxExportBatchSize <= maxQueueSize,
+        "maxExportBatchSize must be smaller or equal to maxQueueSize.");
```

Comment:
Let's omit this for the time being, given @breedx-splk's comment [here](https://github.com/open-telemetry/opentelemetry-java/issues/7024#issuecomment-2617347391):

> Why is this a requirement? Wouldn't it be ok to fill up an export batch with multiple queue drains?